rockcraft/layers.py

```diff
@@ -54,6 +55,32 @@ def archive_layer(
             tar_file.add(filepath, arcname=arcname, recursive=False)
 
 
+def prune_prime_files(prime_dir: Path, files: Set[str], base_layer_dir: Path) -> None:
+    """Remove (prune) files in a prime directory if they exist in the base layer.
+
+    Given a set of filenames ``files``, this function will remove (prune) all those
+    filenames from prime dir ``prime_dir`` if the corresponding sub path exists in
+    the extracted base layer directory ``base_layer_dir`` with same contents and
+    permissions.
+
+    For example, "{prime_dir}/dir/subdir/file1" will be pruned if the matching file
+    "{base_layer_dir}/dir/subdir/file1" exists and has the same contents, owner,
+    group, and permission bits.
```

Comment:
I think you should add the ":param" docstring, so that 1) it is consistent with other docstrings and 2) we can generate docs from it if need be